pkg/user/tx_client.go

```diff
@@ -38,6 +39,14 @@ const (
 
 type Option func(client *TxClient)
 
+// TxResponse is a response from the chain after
+// a transaction has been submitted.
+type TxResponse struct {
+	Height int64
```

Comment:
[question] would it help if this field had a comment that explains the Height will be `0` until the tx gets included on chain?

Reply:
hm, I feel like it's pretty self explanatory but if you think this causes confusion i'm happy to address that as part of my upcoming work on tx client

Reply:
I don't think it's self explanatory for a reader who looks at this struct and doesn't look at implementation details of how the struct is created. The other fields of the struct are initialized when the struct is created but `Height` isn't and is set later. As a reader, I would like to see something like this:

```go
// TxResponse is a response from the chain after
// a transaction has been submitted.
type TxResponse struct {
	// Height is the block height at which the transaction was included on-chain.
	// If the transaction was submitted asynchronously, height is expected to be 0 until it gets included in a block.
	Height int64
	TxHash string
	Code   uint32
}
```

Reply:
The height in TxResponse is set based on the TxStatus response. The height will only be set after that transactions was committed in a block otherwise it will be nil.

Reply:
The default value for an int64 is `0` so it will be `0` not nil. See https://go.dev/play/p/C6yLKG0zTzq

Reply:
my comment was jus referring to the fact that how you submit the transaction does not affect height in this particular case

Reply:
Oh got it, sorry I think you're right that tx client doesn't support an async way to submit txs. Let's resolve this thread, I'm nit-picking about something that shouldn't block this PR.